R/wrangle_and_check.R

```diff
@@ -193,3 +240,44 @@ fill_annual_profit_cols <- function(annual_profits) {
 
   return(annual_profits_filled)
 }
+
+#' Check if values in financial data are plausible
+#'
+#' Checks that:
+#' 1. there is a 1 to 1 mapping of company_name and company_id.
+#' 1. numeric columns hold values in acceptable ranges.
+#'
+#' @inheritParams check_financial_data
+#'
+#' @return NULL
+check_valid_financial_data_values <- function(financial_data, asset_type) {
+  n_unique_ids <- length(unique(financial_data$company_id))
+  n_unique_names <- length(unique(financial_data$company_name))
+  n_unique_id_names_combinations <- length(unique(paste(financial_data$company_id, financial_data$company_name)))
+
+  if (length(unique(c(n_unique_ids, n_unique_names, n_unique_id_names_combinations))) > 1) {
```

Comment:
This implies that users will have to add the correct `company_id` for loans. That is only realistic if they get the company id from their pacta run, how else would they know the id? However P4B does not provide the `company_id` and we return `NA_real_` for all companies when wrangling the P4B results to P4I format. In my understanding this test would fail in that case